### ISSUE TITLE
ErrorHandler showing servlet info, can not be disabled unless overriding most of its functionality

### ISSUE THREAD

**Issue by MatthijsEM** [eclipse/jetty.project#4529]
**Jetty version**
9.4.26.v20200117

**Java version**
Does not matter

**OS type/version**
Does not matter

**Description**
Recently migrated some projects from 9.4.20.v20190813 to 9.4.26.v20200117. The error page response has changed somewhat and now also includes the Servlet name, e.g.

```
HTTP ERROR 404 Not Found
URI: /notfound
STATUS: 404
MESSAGE: Not Found
SERVLET: org.eclipse.jetty.servlet.DefaultServlet-b9a77c8
```

Like the Stacktrace and Powered-By info, which can be disabled, the Servlet parameter contains sensitive information not be exposed to the public. It would be nice if it could easily be disabled. If not, because most of the methods are protected, it can be disabled by overriding all the nice write error methods for the different content types. Maybe the expectation is that as a developer you should override the ErrorPage class anyway if you want customization. If so, feel free to close the ticket.

**Comment by sbordet** [comment]
Fixed in 9.4.27.

**Comment by TobiX** [comment]
The fix looks incomplete: It only removes the servlet from the text and json outputs, not he HTML output...

**Comment by MatthijsEM** [comment]
It appears one was missed. Also needs the `isShowServlet()` over here:

```
htmlRow(writer, "MESSAGE", message);
htmlRow(writer, "SERVLET", request.getAttribute(Dispatcher.ERROR_SERVLET_NAME));
Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
```

https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java#L419

**Comment by joakime** [comment]
> Maybe the expectation is that as a developer you should override the ErrorPage class anyway if you want customization.

This is built into the standard servlet spec.
Just specify a `<error-page>` in your `WEB-INF/web.xml` such as ...

``` xml
<web-app ...>
  <error-page>
    <location>/general-error.html</location>
  <error-page>
</web-app>
```

And make that HTML whatever you want it to be.
Such as an empty file.

**Comment by joakime** [comment]
Fixed in PR #4734
